### PR TITLE
cmd, core, eth: configurable txpool parameters

### PR DESCRIPTION
**Pull request by karalabe**
This PR allows most of the transaction pool parameters to be configured both programatically as well as via CLI flags.

### CODE REVIEW

File: core/tx_pool.go [hunk 1]
```diff
@@ -78,6 +72,48 @@ var (
...
+	Lifetime: 3 * time.Hour,
+}
+
+// sanitize checks the provided user configurations and changes anything that's
+// unreasonable or unworkable.
+func (config *TxPoolConfig) sanitize() TxPoolConfig {
+	conf := *config
+	if conf.PriceLimit < 1 {
+		log.Warn("Sanitizing invalid txpool price limit", "provided", conf.PriceLimit, "updated", 1)
+		conf.PriceLimit = 1
```
Comment (obscuren) [inline core/tx_pool.go]: Why not use `DefaultTxPoolConfig.PriceLimit` here like on line `112`?

Comment (karalabe) [inline core/tx_pool.go]: Hah, fair :) Was thinking about this.

**Comment by karalabe** [comment]
@obscuren Fixed, PTAL

**Comment by holiman** [comment]
Confirmed :+1: 

Running with `--txpool.accountqueue 200` , here's my status after batching in 101 transactions in the wrong order: 
```text

> txpool.status
{
  pending: 101,
  queued: 0
}
```

All of them got into `queued`, then were shoved over to `pending`

**Pull request merged**